x-pack/metricbeat/module/meraki/device_health/uplinks.go

```diff
@@ -0,0 +1,172 @@
+	if err != nil {
+		return fmt.Errorf("GetOrganizationCellularGatewayUplinkStatuses failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+	}
+
+	lossAndLatency, res, err := client.Organizations.GetOrganizationDevicesUplinksLossAndLatency(
+		organizationID,
+		&meraki.GetOrganizationDevicesUplinksLossAndLatencyQueryParams{
+			Timespan: period.Seconds(),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("GetOrganizationDevicesUplinksLossAndLatency failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+	}
+
+	for _, device := range *applicanceUplinks {
+		if device.HighAvailability != nil {
+			devices[Serial(device.Serial)].haStatus = device.HighAvailability
+		}
+
+		if device.Uplinks != nil {
+			var uplinks []*uplink
+			for _, uplinkStatus := range *device.Uplinks {
+				uplink := &uplink{
+					lastReportedAt: device.LastReportedAt,
+					status:         &uplinkStatus,
+				}
+
+				for _, metrics := range *lossAndLatency {
+					if metrics.TimeSeries != nil && metrics.Serial == device.Serial && metrics.Uplink == uplinkStatus.Interface {
+						uplink.lossAndLatency = &metrics
+						break
+					}
+				}
+
+				uplinks = append(uplinks, uplink)
+			}
+
+			devices[Serial(device.Serial)].uplinks = uplinks
+		}
+	}
+
+	for _, device := range *cellularGatewayUplinks {
+		if device.Uplinks == nil {
+			continue
+		}
+
+		var uplinks []*uplink
+		for _, uplinkStatus := range *device.Uplinks {
+			uplink := &uplink{
+				lastReportedAt:        device.LastReportedAt,
+				cellularGatewayStatus: &uplinkStatus,
+			}
+
+			for _, metrics := range *lossAndLatency {
+				if metrics.TimeSeries != nil && metrics.Serial == device.Serial && metrics.Uplink == uplinkStatus.Interface {
+					uplink.lossAndLatency = &metrics
+					break
+				}
+			}
+
+			uplinks = append(uplinks, uplink)
+		}
+
+		devices[Serial(device.Serial)].uplinks = uplinks
+	}
+
+	return nil
+}
+
+func reportUplinkMetrics(reporter mb.ReporterV2, organizationID string, devices map[Serial]*Device) {
+	metrics := []mapstr.M{}
+	for _, device := range devices {
+		if len(device.uplinks) == 0 {
+			continue
+		}
+
+		for _, uplink := range device.uplinks {
+			if uplink.lossAndLatency != nil {
+				// each loss and latency metric can have multiple values per collection.
+				// we report each value as it's own (smaller) metric event, containing
+				// the identifying device/uplink fields.
+				for _, dataPoint := range *uplink.lossAndLatency.TimeSeries {
+					// for some reason there are sometimes empty buckets
+					if dataPoint.LatencyMs != nil || dataPoint.LossPercent != nil {
+						metrics = append(metrics, mapstr.M{
+							"@timestamp":        dataPoint.Ts,
+							"uplink.latency.ms": dataPoint.LatencyMs,
+							"uplink.loss.pct":   dataPoint.LossPercent,
+
+							"device.serial":     uplink.lossAndLatency.Serial,    // _should_ be the same as `device.Serial`
```

Comment:
Was this meant to be `device.serial` or `uplink.serial` ?

Reply:
yeh this is device serial - the reason is that uplinks are not devices, but instead they are connections configured on devices. the unique identifier for an uplink is the device serial and the uplink interface e.g. `Q2BN-59GF-ZZEL/wan1`. this would be the primary WAN ( i.e. ethernet) on an MX device with the given serial number. 

some uplinks

Reply:
Do we add this field in field.yml ?